src/Feldspar/Core/Language.hs

```diff
@@ -444,14 +444,82 @@ err msg = assertMsg msg false undef
 -- FFI.hs
 --------------------------------------------------
 
-{- Omitted
-foreignImport :: ( Type (DenResult a)
-                 , SyntacticN c b
-                 , ApplySym a b FeldDomain
-                 )
-              => String -> Denotation a -> c
-foreignImport name f = sugarSymF (ForeignImport name f)
--}
+-- | Arity specific functions for FFI
+
+foreignImport0 :: (Syntactic r, Type (Internal r))
+               => String -> Internal r -> r
+foreignImport0 s f = sugarSym0 (ForeignImport s (EqBox f))
+
+foreignImport1 :: (Syntactic a, Syntactic r, Type (Internal r), Typeable (Internal a))
```

Comment:
Use `Syntax r` instead of `Syntactic r` + `Type (Internal r)`.

Reply:
Will do!